Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/repository/storage/VaccinationStorage.kt

```diff
@@ -18,75 +18,67 @@ import javax.inject.Singleton
 class VaccinationStorage @Inject constructor(
     @AppContext val context: Context,
     @BaseGson val baseGson: Gson,
-    private val containerPostProcessor: ContainerPostProcessor,
 ) {
     private val mutex = Mutex()
     private val prefs by lazy {
         context.getSharedPreferences("vaccination_localdata", Context.MODE_PRIVATE)
     }
 
     private val gson by lazy {
-        // Allow for custom type adapter.
         baseGson.newBuilder().apply {
-            registerTypeAdapterFactory(containerPostProcessor)
             registerTypeAdapterFactory(CwaCovidCertificate.State.typeAdapter)
         }.create()
     }
 
-    suspend fun load(): Set<VaccinatedPersonData> = mutex.withLock {
+    suspend fun load(): Set<StoredVaccinationCertificateData> = mutex.withLock {
```

Comment:
I found the Set used in RecoveryCertificateRepository not to be helpful as it did not make sure there is only one entry per QR code. The map used in TestcertificateRepository is much better to handle, and ensures there is only one entry per QR code aka containerId

Reply:
I made changes and added test case `test json set with same certificates`